### DISCUSSION TITLE
this.$state in setup store

### DISCUSSION THREAD

**Discussion by ohroy** [vuejs/pinia#2130] [Help and Questions] [answered]
If there are so many prop in store,  what's the simplest way to implement it?

**Comment by posva** [comment, marked as answer]
Unfortunately, it cannot be implemented because it cannot be built until the store setup function is executed. It does exist in methods but cannot be typed in TypeScript.

In the end, this doesn't change much as having state divided into refs is one of the advantages of the composition API, it enables refactoring and grouping features within composables.
